### ISSUE TITLE
blacklistadd failing

### ISSUE THREAD

**Issue by GianniGi** [tridactyl/tridactyl#3853]
Tridactyl version: 1.21.1
Firefox version: Mozilla Firefox 90.0.1
Operating system: win
<!--

# Reporting a bug / getting help

-   Brief description of the problem:
tridactyl reactivates itself on some websites, also when the site had been previously blacklistadd(ed)

-   Steps to reproduce:
:blacklistadd some site
let some time pass (days, random)
tridactyl reactivates itself.
:blacklistadd again, but tridactyl stays active
restart the browser to make the blacklistadd work again

-   Tridactyl version (`:version`):
1.21.1

-   Firefox version (Top right menu > Help > About Firefox):
90.0.1

-   URL of the website the bug happens on:
usually some critical google website, like gmail or google colab

-   Operating system:
windows 10

**Comment by bovine3dom** [comment]
Do you have an RC file with `sanitise` at the top?

When Tridactyl next re-activates itself could you run `:viewconfig --user` and paste it here?

**Comment by GianniGi** [comment]
<html><body>


configversion | "2.0"
-- | --
update |  
lastchecktime | 1598341868200
lastnaggedversion | "1.20.1"
autocmds |  
DocStart |  
https://mail.google.com | "mode ignore"
http://colab.reasearch.google.com | "mode ignore"
https://colab.research.google.com | "mode ignore"
http://colab.research.google.com | "mode ignore"
http://colab.google.com | "mode ignore"
blacklistadd | "https://gmail.com"
hintchars | "fjdkslaqpwoeiruzmxncv"
normal | "https://www.google.com"
subconfigs |  
geekhack.org |  
nmaps |  
f | "hint -c a:not(.navPages), span + a.navPages"
updatenaglastversion | "1.14.6"


</body>
</html>

**Comment by bovine3dom** [comment]
Is Tridactyl currently working OK? That config looks reasonable.

**Comment by GianniGi** [comment]
Yes, it works. The problem is that often tridactyl reactivates itself also on blacklisted pages and this makes no sense... And when this happens, the only thing that you can do is restart the browser.

**Comment by bovine3dom** [comment]
Could you be pressing any of the following keys?

https://github.com/tridactyl/tridactyl/blob/f160b4248596c877fc6cbfd5d689d84b3b21a7f4/src/lib/config.ts#L140-L146

E.g. shift-insert in a text box will put you in insert mode which would then put you in normal mode once you left it.

> the only thing that you can do is restart the browser.

So entering ignore mode by pressing shift-insert doesn't do anything? Reloading the page still puts you in normal mode? Opening a new tab still puts you in normal mode? What does `:viewconfig --user` show in that situation?

**Comment by GianniGi** [comment]
I've pressed the combos that you wrote, but they caused no bit issue, except C-o which is really irritating, it seems close to what I have experienced, since blacklistadd doesn't work and it stays in insert mode. Is there a way to keep a blacklisted page blacklisted, preventing some combos to make everyting go crazy?

**Comment by bovine3dom** [comment]
Maybe this will work better for you:

1. remove all the current blacklisted pages with `blacklistremove [url]`
2. make a new empty mode with, e.g. `:bind --mode=myblacklist <AC-Escape> echo`
3. instead of `:blacklistadd [url]` run `:autocmd DocStart [url] mode myblacklist`

---

In the latest betas or Tridactyl 1.21.2+ (not yet released) you can instead do `seturl [url] superignore true`, which prevents Tridactyl from loading on matching URLs.

**Comment by GianniGi** [comment]
Thank you very much for your work and your answer, I'll probably wait for 1.21.2 since it doesn't happen so frequently.

**Comment by GianniGi** [comment]
For example I have this error showing on a blacklisted page:
# An error occurred in the content controller: TypeError: shadowRoot.activeElement is null

**Comment by bovine3dom** [comment]
That's interesting. I don't think that's anything to do with blacklistadd; it's probably to do with scrolling or focusing.

Which site do you see it on and what are you doing when you see it?

**Comment by GianniGi** [comment]
Colab.research.google.com

No particular action except writing some code


Il ven 23 lug 2021, 13:11 Oliver Blanthorn ***@***.***> ha
scritto:

**Comment by GianniGi** [comment]
It happens daily on google colab https://colab.research.google.com/ 

# An error occurred in the content controller: TypeError: shadowRoot.activeElement is null

**Comment by bovine3dom** [comment]
While you're in insert mode, writing code?

**Comment by GianniGi** [comment]
I don't know, it appeared suddenly while colab was working.

Il mar 3 ago 2021, 15:01 Oliver Blanthorn ***@***.***> ha
scritto:

**Comment by zeorin** [comment]
Hmmm, I have this issue, too.

The log does report that everything works:

**Comment by bovine3dom** [comment]
@zeorin `:unfocus` puts Tridactyl into normal mode which we should probably document :)

https://github.com/tridactyl/tridactyl/blob/fa4a2347189794a96ba5cbf123962be049cd3635/src/excmds.ts#L1337

If you delete those autocmds with autocmddelete it should start working.

**Comment by bovine3dom** [comment]
Thanks, I've updated that :)
